Task15 jdbc dao

andresmadeddie opened the pull request:
Create interfaces, daoFiles, and model for all tables, create connection pools, and add Scheme and SQL scripts files

AndresSolvd committed:
Create interfaces for each table(12) and IDao interface.

Create a testDelete folder and added some temp files to test

AndresSolvd committed:
Create interfaces for each table(12) and IDao interface.

Create a testDelete folder and added some temp files to test

andresmadeddie committed:
Delete comments

andresmadeddie committed:
Add more test files to TestDelete
extend IDao interface to missing interfaces
Modify IDao interface delete method and related fixes
Add a method to look by category name to IDaoCategory
Create Dao file for category table (1/12) and test with main file

andresmadeddie committed:
Create DaoCustomer (2/12)

andresmadeddie committed:
Create DaoCustomer (2/12)
Create DaoOrderItem (3/12)
Create DaoOrders (4/12)
Create DaoOwner (5/12)
Create DaoPerson (6/12)
Create DaoProduct (7/12)
Create DaoPromotion (8/12)
Create DaoStaff (9/12)
Create DaoProductPromotion (10/12)
Create DaoShop (11/12)
Create DaoSupplier (12/12)

andresmadeddie committed:
Create DaoCustomer (2/12)
Create DaoOrderItem (3/12)
Create DaoOrders (4/12)
Create DaoOwner (5/12)
Create DaoPerson (6/12)
Create DaoProduct (7/12)
Create DaoPromotion (8/12)
Create DaoStaff (9/12)
Create DaoProductPromotion (10/12)
Create DaoShop (11/12)
Create DaoSupplier (12/12)

andresmadeddie committed:
change names of all JDBC directory

andresmadeddie committed:
change names of all JDBC directory

andresmadeddie committed:
add Scheme and SQL scripts files into resources folder

ankochergin commented on src/main/java/com/solvd/sql/interfaces/IDao.java:
```diff
@@ -8,15 +8,15 @@
     // CRUD Create
```
rename interface to IBaseDAO

ankochergin commented on src/main/java/com/solvd/sql/interfaces/IDao.java:
```diff
@@ -8,15 +8,15 @@
     List<T> getAll() throws SQLException;
+
+    // CRUD Read
+    T get(int id) throws SQLException;
```
```suggestion
    T getById(int id) throws SQLException;
```

ankochergin commented on src/main/java/com/solvd/sql/interfaces/IDaoOrders.java:
```diff
@@ -0,0 +1,6 @@
+
+import com.solvd.sql.model.Orders;
+
+public interface IDaoOrders extends IDao<Orders> {
```
if you don't have any method here so make your OrdersDAO implementing IBaseDao<Orders> directly

ankochergin replied:
the same for other DAO interfaces

ankochergin commented on src/main/java/com/solvd/sql/jdbc/CustomerDao.java:
```diff
@@ -0,0 +1,111 @@
+public class CustomerDao implements IDaoCustomer {
+    private final ConnectionPool connectionPool = ConnectionPool.getInstance();
+    @Override
+    public void insert(Customer customer) throws SQLException {
```
```suggestion
public class CustomerDao implements IDaoCustomer {

    private final ConnectionPool connectionPool = ConnectionPool.getInstance();
    
    @Override
    public void insert(Customer customer) throws SQLException {
```

ankochergin replied:
update this formatting in all classes

ankochergin commented on src/main/java/com/solvd/Main.java:
```diff
@@ -1,31 +1,32 @@
-            int oid = rs.getInt("id");
-            System.out.println(oid);
-        }
+        // CATEGORY
```
have you tested all tables?

andresmadeddie committed:
fix route

andresmadeddie committed:
Create XML folder with the xml, xds, and a DOMParser file

andresmadeddie committed:
Rename interface to IBaseDAO
Implement IBaseDAO on all tables that does not have further methods to implement
adding lines to each classes to comply with the required format

andresmadeddie committed:
Rename interface to IBaseDAO
Implement IBaseDAO on all tables that does not have further methods to implement
adding lines to each classes to comply with the required format

andresmadeddie committed:
Fix index in CustomerDao on Insert method, eliminate a comma in the Update method
Fix index in StaffDao update method
Fix index in SupplierDao update method
Fix index in PromotionDao insert and update method
Fix index in ProductPromotionDao insert and update method. delete an extra comma  in update method

Create queries on Main Class to test all Tables methods

jinocencio-solvd commented on src/main/java/com/solvd/Main.java:
```diff
@@ -1,31 +1,344 @@
+        // CATEGORY
+        System.out.println("\n--- CATEGORY ---\n");
+        // Create
+        CategoryDao categoryDao = new CategoryDao();
```
call your DAOs using a service layer

jinocencio-solvd commented on src/main/java/com/solvd/Main.java:
```diff
@@ -1,31 +1,344 @@
+        // Delete
+        personDao.delete(5);
+        // Read
+        System.out.println(personDao.get(4));
```
verify same person by implementing hashcode and equals in the person class and for each of your models if applicable

jinocencio-solvd commented on src/main/java/com/solvd/util/XML/shop.xml:
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
XML files should be in resources, DOMParser.java is okay since it's a java file

andresmadeddie replied:
This will be handle in a new branch

jinocencio-solvd commented on src/main/java/com/solvd/Main.java:
```diff
@@ -1,31 +1,344 @@
-import java.util.List;
-
 public class Main {
     public static void main(String[] args) throws SQLException {
```
Better to handle  SQLException than have main throw it

ankochergin commented on src/main/java/com/solvd/sql/interfaces/IDaoCategory.java:
```diff
@@ -0,0 +1,9 @@
+
+import java.sql.SQLException;
+
+public interface IDaoCategory extends IBaseDAO<Category> {
```
```suggestion
public interface ICategoryDao extends IBaseDAO<Category> {
```

ankochergin commented on src/main/java/com/solvd/util/XML/DOMParser.java:
```diff
@@ -0,0 +1,35 @@
+import javax.xml.parsers.*;
+import java.io.*;
+
+public class DOMParser {
```
could you please remove XML hw from this PR? lets create a new separate one for XML after this one is merged

ankochergin commented on src/main/java/com/solvd/util/XML/DOMParser.java:
```diff
@@ -0,0 +1,35 @@
+package com.solvd.util.XML;
```
```suggestion
package com.solvd.util.xml;
```

andresmadeddie replied:
I did change it and store it somewhere else for a new branch. But all the XML folder was deleted from this branch

andresmadeddie committed:
change interface name to ICategoryDao
remove XML folder
added try and catch for SQL Exceptions in Main Class and remove throw

andresmadeddie committed:
Rename Interfaces

andresmadeddie committed:
Create Services and modify Main class accordingly

andresmadeddie committed:
Eliminate unused import statement

ankochergin commented on src/main/java/com/solvd/sql/interfaces/ICustomerDao.java:
```diff
@@ -0,0 +1,6 @@
+import com.solvd.sql.model.Customer;
+
+public interface ICustomerDao extends IBaseDAO<Customer> {
+}
```
if interface is empty just make `CustomerDao implements IBaseDAO<Customer>` and remove empty interface

ankochergin commented on src/main/java/com/solvd/sql/interfaces/IBaseDAO.java:
```diff
@@ -3,20 +3,20 @@
+
+    // CRUD Read
+    T get(int id) throws SQLException;
 }
```
remove `throws SQLException;` in all methods and handle all exceptions using try-catch in DAOs

ankochergin commented on src/main/java/com/solvd/Main.java:
```diff
@@ -1,31 +1,348 @@
-        Connection conn = TestConnection.getConnection();
+    public static void main(String[] args) {
+
+        try {
```
handle all SQLException exceptions in DAOs

AndresSolvd committed:
Change method to GetById, and GetByName

AndresSolvd committed:
Change method to GetById, and GetByName
Remove the Throw SQL Exception for Interface and customerService. All SQL Exception is managed inside Dao File methods
delete unused Interfaces

AndresSolvd committed:
remove unused imports

AndresSolvd committed:
remove unused imports
remove throw in main class
